cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/existingdataproc/ExistingDataprocConf.java

```diff
@@ -0,0 +1,291 @@
+   * @throws IOException if there was an error reading the account key
+   */
+  GoogleCredential getComputeCredential() throws IOException {
+    if (accountKey == null) {
+      return GoogleCredential.getApplicationDefault();
+    }
+
+    try (InputStream is = new ByteArrayInputStream(accountKey.getBytes(StandardCharsets.UTF_8))) {
+      return GoogleCredential.fromStream(is)
+        .createScoped(Collections.singleton(CLOUD_PLATFORM_SCOPE));
+    }
+  }
+
+  /**
+   * @return GoogleCredentials for use with Dataproc
+   * @throws IOException if there was an error reading the account key
+   */
+  GoogleCredentials getDataprocCredentials() throws IOException {
+    if (accountKey == null) {
+      return getComputeEngineCredentials();
+    }
+
+    try (InputStream is = new ByteArrayInputStream(accountKey.getBytes(StandardCharsets.UTF_8))) {
+      return GoogleCredentials.fromStream(is).createScoped(CLOUD_PLATFORM_SCOPE);
+    }
+  }
+
+  private static GoogleCredentials getComputeEngineCredentials() throws IOException {
+    try {
+      GoogleCredentials credentials = ComputeEngineCredentials.create();
+      credentials.refreshAccessToken();
+      return credentials;
+    } catch (IOException e) {
+      throw new IOException("Unable to get credentials from the environment. "
+                              + "Please explicitly set the account key.", e);
+    }
+  }
+
+  private String getMachineType(int cpus, int memoryGB) {
+    // TODO: there are special names for pre-defined cpu and memory
+    // for example, 4cpu 3.6gb memory is 'n1-highcpu-4', 4cpu 15gb memory is 'n1-standard-4'
+    return String.format("custom-%d-%d", cpus, memoryGB);
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation.
+   *
+   * @throws IllegalArgumentException if it is an invalid config
+   */
+  static ExistingDataprocConf fromProperties(Map<String, String> properties) {
+    return create(properties);
+  }
+
+  static ExistingDataprocConf create(Map<String, String> properties) {
+    String accountKey = getString(properties, "accountKey");
+    if (accountKey == null || AUTO_DETECT.equals(accountKey)) {
+      try {
+        getComputeEngineCredentials();
+      } catch (IOException e) {
+        throw new IllegalArgumentException(e.getMessage(), e);
+      }
+    }
+    String projectId = getString(properties, PROJECT_ID_KEY);
+    if (projectId == null || AUTO_DETECT.equals(projectId)) {
+      projectId = getSystemProjectId();
+    }
+    String region = getString(properties, "region");
+    if (region == null || AUTO_DETECT.equals(region)) {
+      region = getRegionFromZone(getSystemZone());
+    }
+    String clustername = getString(properties, "clusterName");
+    if (clustername == null) {
+      throw new IllegalArgumentException("Dataproc ClusterName is mandatory.");
+    }
+
+    boolean sshMonitoringEnabled = Boolean.parseBoolean(properties.get("sshMonitoringEnabled"));
```

Comment:
Shouldn't take it from properties. Use the `ProvisionerContext.getRuntimeMonitorType()` method

Reply:
Discussing with Vinisha and Bhooshan we need the ssh for Sandbox testing and hence it was introduced.   If this is automatically set to the right value depending on the deployment then I will modify it accordingly and remove the option from configuration.

Reply:
The `ProvisionerContext.getRuntimeMonitorType()` is the one that will provide the correct value in all environments.

Reply:
As we discussed different environments will use different monitoring ways. We should make sure implementation works for all the environments.

Reply:
During the profile creation using UI , the validate properties method is invoked which has an argument of all the properties from the UI. During this stage I need to validate  if its SSH then ensure the user and cert is provided.   In this method I dont have access to the provisioner context and wondering is there a different way to check it so I can ensure this validation succeeds.

Reply:
`sshMonitoringEnabled` UI property seem to be redundant. Can just have ssh user and ssh key optional properties in UI and can get rid of `sshMonitoringEnabled`.
As terence has suggested, in createCluster, we can rely on `ProvisionerContext.getRuntimeMonitorType()` to choose what monitoring should be used.

Reply:
Currently if its SSH then I check if sshuser and cert is provided else I throw a error, with this approach I will not be able to do that. It will be only at the time of create cluster that I will be able to check. Is this acceptable?